server.js

```diff
@@ -168,15 +167,14 @@ app.use(sessionMiddleware(app));
 app.use(passportMiddleware());
 app.use(redirectsMiddleware.common);
 app.use(localsMiddleware.middleware);
+app.use(previewMiddleware);
```

Comment:
This was the key change, preview middleware needs to come _after_ passport middleware otherwise you get infinite redirects.